src/integrations/watchers/indexable-home-page-watcher.php

```diff
@@ -108,6 +122,11 @@ public function check_option( $old_value, $new_value, $option ) {
 	 */
 	public function build_indexable() {
 		$indexable = $this->repository->find_for_home_page( false );
+
+		if ( $indexable === false && ! $this->indexable_helper->should_index_indexables() ) {
+			return;
+		}
+
 		$indexable = $this->builder->build_for_home_page( $indexable );
 
 		if ( $indexable ) {
```

Comment:
Underneath here we do $indexable->save() I am wondering if we should for super consistency sake also use the indexable helper here to call ->save_indexable() or was there a reason not to.

Reply:
Just to append here. I see the same in the `src/integrations/watchers/indexable-static-home-page-watcher.php` File.

Reply:
Yeah, I get what you mean. This was a conscious choice, and I describe the reasons in the technical choices:
>Decided to keep a couple of `$indexable->save()`s intact, in the indexable homepage watcher and the indexable static homepage watcher, because they are only triggered if there's an indexable already there, which then gets updated. The reasoning is that:
> * it doesn't create a new indexable
> * it actually makes sense that if there is an existing indexable, to update its values with the new stuff, to avoid having stale data, which might be a problem when/if the filter gets removed at some point.
> * for the indexable homepage watcher specifically, I had to [some logic](https://github.com/Yoast/wordpress-seo/pull/21335/files#diff-a454493beeb1d4945dcab5c5fa629faf5f92b7a55ccbe044ae297d65f2151b67R126) to make sure that the `$indexable->save()` does only updates and not creates.

But let's talk, because I'm definitely not 100% sure I took the right path here :)

Reply:
I think this is fine in this scenario :)